pimd/pim_cmd.c

```diff
@@ -6413,52 +6487,107 @@ static void show_mroute_summary(struct pim_instance *pim, struct vty *vty)
 		}
 	}
 
-	vty_out(vty, "%-20s %d/%d\n", "(*, G)", starg_hw_mroute_cnt,
-		starg_sw_mroute_cnt + starg_hw_mroute_cnt);
-	vty_out(vty, "%-20s %d/%d\n", "(S, G)", sg_hw_mroute_cnt,
-		sg_sw_mroute_cnt + sg_hw_mroute_cnt);
-	vty_out(vty, "------\n");
-	vty_out(vty, "%-20s %d/%d\n", "Total",
-		(starg_hw_mroute_cnt + sg_hw_mroute_cnt),
-		(starg_sw_mroute_cnt +
-		 starg_hw_mroute_cnt +
-		 sg_sw_mroute_cnt +
-		 sg_hw_mroute_cnt));
+	if (!json) {
+		vty_out(vty, "%-20s %u/%u\n", "(*, G)", starg_hw_mroute_cnt,
+			starg_sw_mroute_cnt + starg_hw_mroute_cnt);
+		vty_out(vty, "%-20s %u/%u\n", "(S, G)", sg_hw_mroute_cnt,
+			sg_sw_mroute_cnt + sg_hw_mroute_cnt);
+		vty_out(vty, "------\n");
+		vty_out(vty, "%-20s %u/%u\n", "Total",
+			(starg_hw_mroute_cnt + sg_hw_mroute_cnt),
+			(starg_sw_mroute_cnt + starg_hw_mroute_cnt
+			 + sg_sw_mroute_cnt + sg_hw_mroute_cnt));
+	} else {
+		/* (*,G) route details */
+		json_starg = json_object_new_object();
+		json_object_object_add(json, "(*,G)", json_starg);
+
+		json_object_int_add(json_starg, "installed",
+				    starg_hw_mroute_cnt);
+		json_object_int_add(json_starg, "total",
+				    starg_sw_mroute_cnt + starg_hw_mroute_cnt);
+
+		/* (S, G) route details */
+		json_sg = json_object_new_object();
+		json_object_object_add(json, "(S,G)", json_sg);
+
+		json_object_int_add(json_sg, "installed", sg_hw_mroute_cnt);
+		json_object_int_add(json_sg, "total",
+				    sg_sw_mroute_cnt + sg_hw_mroute_cnt);
+
+		json_object_int_add(json, "totalNumOfInstalledMroutes",
+				    starg_hw_mroute_cnt + sg_hw_mroute_cnt);
+		json_object_int_add(json, "totalNumOfMroutes",
+				    starg_sw_mroute_cnt + starg_hw_mroute_cnt
+					    + sg_sw_mroute_cnt
+					    + sg_hw_mroute_cnt);
+	}
 }
 
 DEFUN (show_ip_mroute_summary,
        show_ip_mroute_summary_cmd,
-       "show ip mroute [vrf NAME] summary",
+       "show ip mroute [vrf NAME] summary [json]",
        SHOW_STR
        IP_STR
        MROUTE_STR
        VRF_CMD_HELP_STR
-       "Summary of all mroutes\n")
+       "Summary of all mroutes\n"
+       JSON_STR)
 {
 	int idx = 2;
+	bool uj = use_json(argc, argv);
 	struct vrf *vrf = pim_cmd_lookup_vrf(vty, argv, argc, &idx);
+	json_object *json = NULL;
+
+	if (uj)
+		json = json_object_new_object();
 
 	if (!vrf)
 		return CMD_WARNING;
 
-	show_mroute_summary(vrf->info, vty);
+	show_mroute_summary(vrf->info, vty, json);
+
+	if (uj) {
+		vty_out(vty, "%s\n", json_object_to_json_string_ext(
+					     json, JSON_C_TO_STRING_PRETTY));
```

Comment:
use clang-format

Reply:
clang-formatter already applied on this patch. clang-formtter is formatting this way.